weaviate/users/user_operations_db.go

```diff
@@ -0,0 +1,43 @@
+package users
+
+import "github.com/weaviate/weaviate-go-client/v5/weaviate/connection"
+
+type UserOperationsDB struct {
+	connection *connection.Connection
+}
+
+func (r *UserOperationsDB) RolesGetter() *UserRolesGetter {
+	return (&UserRolesGetter{connection: r.connection}).WithUserType(UserTypeDB)
+}
+
+func (r *UserOperationsDB) RolesAssigner() *RoleAssigner {
+	return (&RoleAssigner{connection: r.connection}).WithUserType(UserTypeDB)
```

Comment:
A user can still override the user type via WithUserType later -- is that an issue?
This lets us keep code duplication at a minimum and you can never make stuff entirely fool-proof; could we make `withUserType()` an unexported method?